res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,417 @@
+var NumarkScratch = {};
+
+/*
+ * USER CONFIGURABLE SETTINGS.
+ */
+
+// Defines the Beatloop Roll sizes for the 4 pads, for available values see:
+// https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatlooproll_X_activate
+// Default: [0.25, 0.5, 1, 2]
+NumarkScratch.autoLoopSizes = [
+    "0.25",
+    "0.5",
+    "1",
+    "2",
+];
+
+// Defines how the Loop Encoder functions.
+// If 'true' the Encoder scrolls the library/loads track. Shift + Encoder manages looping.
+// If 'false' the Encoder manages looping. Shift + Encoder scrolls the library/loads track (Serato default).
+// Default: false
+NumarkScratch.invertLoopEncoderFunction = false;
+
+// Defines the bightness of button LEDs when they inactive.
+// '0x00' sets the LEDSs to completely off. '0x01 sets the LEDs to dim.
+// Default: 0x01 (dim)
+NumarkScratch.LOW_LIGHT = 0x01;
+
+/*
+ * CODE
+ */
+
+
+components.Button.prototype.off = NumarkScratch.LOW_LIGHT;
+
+NumarkScratch.init = function() {
+    // Initialize component containers
+    NumarkScratch.deck = new components.ComponentContainer();
+    NumarkScratch.effect = new components.ComponentContainer();
+    for (let i = 0; i < 2; i++) {
+        NumarkScratch.deck[i] = new NumarkScratch.Deck(i + 1);
+        NumarkScratch.effect[i] = new NumarkScratch.EffectUnit(i + 1);
+    }
+
+    NumarkScratch.xfader = new NumarkScratch.XfaderContainer();
+
+    const createVuCallback = function(deckOffset) {
+        return function(value) {
+            midi.sendShortMsg(0xB0 + deckOffset, 0x1F, value * 90);
+        };
+    };
+
+    engine.makeConnection("[Channel1]", "vu_meter", createVuCallback(0));
+    engine.makeConnection("[Channel2]", "vu_meter", createVuCallback(1));
+
+    // Trigger is needed to initialize lights to 0x01
+    NumarkScratch.deck.forEachComponent(function(component) {
+        component.trigger();
+    });
+    NumarkScratch.effect.forEachComponent(function(component) {
+        component.trigger();
+    });
+
+    NumarkScratch.shiftButton = new components.Button({
+        midi: [0x9F, 0x32],
+        input: function(channel, control, value) {
+            if (this.isPress(channel, control, value)) {
+                NumarkScratch.shift();
+                this.send(this.on);
+            } else {
+                NumarkScratch.unshift();
+                this.send(this.off);
+            }
+        },
+        trigger: function() {
+            this.send(this.off);
+        }
+    });
+
+    NumarkScratch.shiftButton.trigger();
+
+    // set FX buttons init light & Shift button init light
+    midi.sendShortMsg(0x98, 0x00, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x01, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x02, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x03, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x04, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x05, NumarkScratch.LOW_LIGHT);
```

Comment:
are you sure this is necessary anymore now that do 
```js
    NumarkScratch.effect.forEachComponent(function(component) {
        component.trigger();
    });
```
above?

Reply:
Not needed. removed. Thanks